vaadin-map-flow-parent/vaadin-map-testbench/src/main/java/com/vaadin/flow/component/map/testbench/MapElement.java

```diff
@@ -38,8 +38,10 @@ public void clickAtCoordinates(double x, double y) {
                 "return arguments[0].configuration.getPixelFromCoordinate([arguments[1], arguments[2]])",
                 this, x, y);
 
-        int clickX = startLeft + pixelCoordinates.get(0).intValue();
-        int clickY = startTop + pixelCoordinates.get(1).intValue();
+        int clickX = startLeft
+                + (int) Math.round(pixelCoordinates.get(0).doubleValue());
+        int clickY = startTop
+                + (int) Math.round(pixelCoordinates.get(1).doubleValue());
```

Comment:
With all coordinates now being run through a conversion I ran into some JS number precision issue with pixel values (which are returned as a fractional number on the client). Applied a rounding, otherwise converting into an int would just truncate the number.